core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1430,14 +1430,10 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
       schemeCorrectedPath
     }
     val timestamp = System.currentTimeMillis
-    addedFiles(key) = timestamp
-
-    // Fetch the file locally in case a job is executed using DAGScheduler.runLocally().
-    Utils.fetchFile(path, new File(SparkFiles.getRootDirectory()), conf, env.securityManager,
```

Comment:
I believe that this line is unnecessary now that `runLocally` was removed a few releases ago. However, I suppose that we might need it in order to handle the somewhat obscure corner-case where a user's `reduce` function / closure executes on the driver and accesses the `SparkFiles` object.